sqlserver/tests/test_statements.py

```diff
@@ -337,6 +353,43 @@ def _obfuscate_sql(sql_query, options=None):
     assert metric['dd_tables'] == expected_metadata_payload['tables']
     assert metric['dd_commands'] == expected_metadata_payload['commands']
 
+    # Setup activity for metadata testing
+    blocking_query = "INSERT INTO ϑings WITH (TABLOCK, HOLDLOCK) (name) VALUES ('puppy')"
+    fred_conn = _get_conn_for_user('fred')
+
+    def _run_query_with_mock_obfuscator(conn, query):
+        # Execute the query with the mocked obfuscate_sql. The result should produce an event payload with the metadata.
+        with mock.patch.object(datadog_agent, 'obfuscate_sql', passthrough=True) as mock_agent:
+            mock_agent.side_effect = _obfuscate_sql
+            _run_test_query(conn, query)
+
+    # bob's query blocks until the tx is completed, so it needs to be run asynchronously.
+    executor = concurrent.futures.ThreadPoolExecutor(1)
+    executor.submit(_run_query_with_mock_obfuscator, bob_conn, blocking_query)
+
+    # fred's query will get blocked by bob, so it needs to be run asynchronously.
+    executor.submit(_run_query_with_mock_obfuscator, fred_conn, query)
+
+    dd_run_check(check)
+
+    bob_conn.commit()
+    bob_conn.close()
+    fred_conn.close()
+    executor.shutdown(wait=True)
+
+    dbm_activity = aggregator.get_event_platform_events("dbm-activity")
```

Comment:
any reason this test isn't in the activity tests file?

Reply:
@jmeunier28 Yeah, I was torn between whether this should go in the activity test file and copy pasta or to continue testing metadata in one place for FQT, metrics, samples, and activity. No hard preference, so if it makes more sense to move it, happy to do so.

Reply:
IDK, It feels like it should since the code change is in `activity.py`, but I don't feel super strongly about it. it might be confusing for example, if a code change in `activity.py` caused the tests in `test_statements.py` to fail, right?